yt_dlp/extractor/radlive.py

```diff
@@ -0,0 +1,406 @@
+                    lrn
+                  }
+                }
+
+
+                fragment Season on Season {
+                  id
+                  lrn
+                  title
+                  acl
+                  number
+                  summary
+                  short_summary
+                  assets
+                  structured_data
+                }
+
+
+                fragment Episode on Episode {
+                  id
+                  lrn
+                  title
+                  acl
+                  assets
+                  number
+                  summary
+                  short_summary
+                  structured_data
+                  associated_miniseries {
+                    lrn
+                  }
+                  associated_season {
+                    lrn
+                    associated_series {
+                      lrn
+                    }
+                  }
+                }
+
+
+                fragment Stream on Stream {
+                  id
+                  lrn
+                  title
+                  short_title
+                  summary
+                  short_summary
+                  acl
+                  type
+                  assets
+                  structured_data
+                }
+
+
+                fragment Playlist on Playlist {
+                  id
+                  lrn
+                  title
+                  short_title
+                  short_summary
+                  summary
+                  acl
+                  assets
+                  associated_channels {
+                    lrn
+                  }
+                }
+
+
+                fragment Metadata on Metadata {
+                    name
+                    lrn
+                    marketing_body_text
+                    marketing_header_text
+                }
+
+
+                fragment Presentation on Presentation {
+                    avails_open_date
+                    avails_close_date
+                    lrn
+                    note
+                    condition
+                    is_released
+                    is_opened
+                    is_closed
+                    restrictions {
+                      enabled
+                      lrn
+                    }
+                }
+                """
+            ),
+            'variables': {
+                'lrn': f'lrn:12core:media:content:channel:{channel_id}',
+            },
+        }).encode('utf-8'))
+        req.add_header('Content-Type', 'application/json')
+        data = json.loads(urllib.request.urlopen(req).read())
+
+        entries = []
+        for feature in data['data']['channel']['features']:
+            for video in feature['assets']['videos']:
+                entries.append({
+                    '_type': 'url_transparent',
+                    'url': feature['structured_data']['url'],
+                    'ie_key': RadLiveIE.ie_key(),
+                })
```

Comment:
why the inner loop? did u mean `video['structured_url']['url']`?
if so,
```suggestion
        entries = [
            self.url_result(video['structured_data']['url'], ie_key=RadLiveIE.ie_key())
            for video in traverse_obj(data, ('data', 'channel', 'features', ..., 'assets', 'videos')]
```
(yes, the `...` is literal)

It would be useful to have id here too

PS: since this is not easily doable with `try_get`, might as well use `traverse_obj` everywhere

Reply:
Technically, `videos` only had one entry in each case I tested. I did a for loop just in case, but not sure if it even makes sense.

Reply:
I see what you mean now. I'll keep it as `feature['structured_data']['url']` as that one returns the actual watch URL so it can pass on to RadLiveIE to extract more info. `video['url']` is the direct m3u8 URL.